### ISSUE TITLE
Desktop icon jiggles when hiding panels to left or right

### ISSUE THREAD

**Issue by MingcongBai** [mate-desktop/caja#942]
#### Expected behaviour

Using "hide buttons" on either side of the panel, the panel should hide to the left or right edge of the screen smoothly while not affecting desktop contents.

#### Actual behaviour

When doing so, the desktop icons "jerks" to the left or right rapidly, while everything else works as expected.

#### Steps to reproduce the behaviour

Create a full width panel, and enable "show hide button" in panel properties. From there, click the hide buttons.

#### MATE general version

1.20.0.

#### Package version

mate-*: 1.20.0.

#### Linux Distribution

AOSC OS

#### Link to downstream report of your Distribution

N/A

**Comment by lukefromdc** [comment]
I just tested this, and here's what I found. Making a new panel or deleting a panel in any position moves the desktop, as the panel's "struts" reserve space for it. On top, bottom,and right side  panels, the panel struts must still be there when the panel is hidden manually or is autohidden, as the desktop stays in places as the panel is hidden or unhidden.

Panels on the left are are another story: the desktop moves as they are manually hidden, but does not move if they are set to autohide. Not only that, toggling autohide on and off also moves the desktop. Apparently something happens to the struts on a left side panel if the panel is manually hidden, or if autohide is enabled at all
